client/serviceregistration/mock/mock.go

```diff
@@ -0,0 +1,125 @@
+package mock
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/client/serviceregistration"
+)
+
+// Ensure that the mock handler implements the service registration handler
+// interface.
+var _ serviceregistration.Handler = (*ServiceRegistrationHandler)(nil)
+
+// ServiceRegistrationHandler is the mock implementation of the
+// serviceregistration.Handler interface and can be used for testing.
+type ServiceRegistrationHandler struct {
+	log hclog.Logger
+
+	// ops tracks the requested operations by the caller during the entire
+	// lifecycle of the ServiceRegistrationHandler. The mutex should be used
+	// whenever interacting with this.
+	ops []Operation
+	mu  sync.Mutex
```

Comment:
In the most extreme of nitpicks, conventionally the lock goes on top of things it locks